src/ralph/lib/polymorphic/models.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+
+from itertools import groupby
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.db.models.fields.related import OneToOneField
+
+from ralph.admin.sites import ralph_site
+
+
+class PolymorphicQuerySet(models.QuerySet):
+
+    def iterator(self):
+        """
+        Override iterator:
+            - Iterate for all objects and collected ID
+            - For each ContentType generates additional queryset
+            - Returns iterator with different models
+        """
+        result = []
+        ct_map = {}
+        for obj in super().iterator():
+            ct_map[obj.content_type_id] = obj.content_type.model_class()
+            result.append((
+                obj.content_type_id, obj.pk)
+            )
+        result = sorted(result, key=lambda x: x[0])
+        result = groupby(result, lambda x: x[0])
+
+        result_query = []
+        for k, v in result:
+            model = ct_map[k]
+            polymorphic_model = getattr(model, 'polymorphic_model')
```

Comment:
the same case as above - we should somehow (maybe from `__mro__`?) deduce what are base classes (classes, which are inherited) and what are final classes (classes, which are not inherited anymore).